### PR TITLE
feat(next): remove CCC

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Closes PLT-2551

## Testing

Should still pass

## Docs

https://github.com/withastro/docs/pull/9507

### CODE REVIEW

File: .changeset/wise-carrots-float.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+'astro': minor
```
Comment (florian-lefebvre) [inline .changeset/wise-carrots-float.md:2]: IDK if that should be considered a major, since we can in theory remove experimental flags at any time

**Comment by florian-lefebvre** [comment]
I may have removed too much stuff according to the smoke tests :thinking:, idk what tho

File: .changeset/wise-carrots-float.md (hunk 1 above)
Comment (natemoo-re) [reply .changeset/wise-carrots-float.md:2]: Minor seems OK to me!

File: .changeset/wise-carrots-float.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+To migrate, remove the flag from your Astro config:
+
+```diff
+export default defineConfig({
+    experimental: {
+-        contentCollectionsCache: true
+    }
+})
+```
```
Comment (natemoo-re) [inline .changeset/wise-carrots-float.md:15]: The replacement for this is migrating to the Content Layer, right? It'd be nice to include that reasoning in the changelog so users have all the context.

**Review by natemoo-re — COMMENTED** [review]
RIP. Code changes all look good to me!

File: .changeset/wise-carrots-float.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+'astro': minor
+---
+
+Removes the experimental `contentCollectionsCache` introduced in `3.5.0`.
```
Comment (sarah11918) [inline .changeset/wise-carrots-float.md:5]: Just making sure we also have a note to add this to the list of experimental flags removed in the v5 upgrade guide!

**Review by bluwy — COMMENTED** [review]
`internals.cacheManifestUsed` and `internals.cachedClientEntries` can also be removed.

File: .changeset/wise-carrots-float.md (hunk 3 above)
Comment (florian-lefebvre) [reply .changeset/wise-carrots-float.md:5]: https://github.com/withastro/docs/pull/9507

**Review by bluwy — APPROVED** [review]

File: .changeset/wise-carrots-float.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Astro Content Layer is now stable, and solves some of the issues of the legacy Content Collections. A Content Layer Cache may be introduced in the future but in the meantime, remove the flag from your Astro config:
```
Comment (sarah11918) [inline .changeset/wise-carrots-float.md:7]:
```suggestion
Astro Content Layer API independently solves some of the caching and performance issues with legacy content collections that this strategy attempted to address.  This feature has been replaced with continued work on improvements to the content layer. If you were using this experimental feature, you must now remove the flag from your Astro config as it no longer exists:
```
Maybe better not to mention things that may or may not happen?

**Pull request merged**